app/script/telemetry/calling/FlowTelemetry.coffee

```diff
@@ -170,38 +170,39 @@ class z.telemetry.calling.FlowTelemetry
 
   # Update statics for the last time and then reset them and the polling interval.
   reset_statistics: =>
-    return new Promise (resolve, reject) =>
-      @_update_statistics()
-      .then =>
-        resolve @statistics
-        @statistics = {}
-      .catch (error) =>
-        @logger.warn "Failed to update flow networks stats: #{error.message}"
-        reject error
-      window.clearInterval @stats_poller
-      @stats_poller = undefined
+    return unless @stats_poller
+
+    window.clearInterval @stats_poller
+    @stats_poller = undefined
+
+    @_update_statistics()
+    .then =>
+      @logger.info 'Network stats updated for the last time', @statistics
+      amplify.publish z.event.WebApp.DEBUG.UPDATE_LAST_CALL_STATUS, @create_report()
+      @statistics = {}
 
   ###
   Start statistics polling.
   @param ice_connection_state [RTCIceConnectionState] Current state of ICE connection
   ###
   start_statistics: =>
-    if not @stats_poller
-      # Track call stats
-      @time_step z.telemetry.calling.CallSetupSteps.ICE_CONNECTION_CONNECTED
-      $.extend @statistics, new z.telemetry.calling.ConnectionStats()
-      @connected()
-
-      # Report calling stats within specified interval
-      window.setTimeout =>
-        @_update_statistics()
-        .then => @logger.info 'Flow network stats updated for the first time', @statistics
-        .catch (error) => @logger.warn "Failed to update flow networks stats: #{error.message}"
-      , 50
-      @stats_poller = window.setInterval =>
-        @_update_statistics()
-        .catch (error) => @logger.warn "Flow networks stats not updated: #{error.message}"
-      , 2000
+    return if @stats_poller
+
+    # Track call stats
+    @time_step z.telemetry.calling.CallSetupSteps.ICE_CONNECTION_CONNECTED
+    $.extend @statistics, new z.telemetry.calling.ConnectionStats()
+    @connected()
+
+    # Report calling stats within specified interval
+    window.setTimeout =>
+      @_update_statistics()
+      .then => @logger.info 'Network stats updated for the first time', @statistics
+      .catch (error) => @logger.warn "Failed to update flow networks stats: #{error.message}"
+    , 50
+    @stats_poller = window.setInterval =>
+      @_update_statistics()
+      .catch (error) => @logger.warn "Networks stats not updated: #{error.message}"
+    , 2000
```

Comment:
It's not part of this PR.. but would be nice to start getting rid of the magic numbers 🎆